src/Catalog/Registration/FlatContainerPackagePathProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Registration
+{
+    public class FlatContainerPackagePathProvider : IPackagePathProvider
+    {
+        public string GetPackagePath(string id, string version)
+        {
+            version = NuGetVersionUtility.NormalizeVersion(version);
+
+            return $"v3-flatcontainer/{id.ToLowerInvariant()}/{version.ToLowerInvariant()}/{id.ToLowerInvariant()}.{version.ToLowerInvariant()}.nupkg";
```

Comment:
We came to the conclusion that it would be best not to use the V3 `index.json` because it's now important to have the registration blobs use the same flat-container, and using the `index.json` could lead to a mismatch.